cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -5,354 +5,1089 @@
 #include <bsg_manycore_elf.h>
 #include <bsg_manycore_mem.h>
 #include <bsg_manycore_loader_dep.h>
+#include <bsg_manycore_loader.h>
+#include <bsg_manycore.h>
+#include <bsg_manycore_printing.h>
+#include <bsg_manycore_eva.h>
+#include <bsg_manycore_origin_eva_map.h>
 
-static const uint32_t KERNEL_REG = 0x1000 >> 2; //!< EPA of kernel. 
-static const uint32_t ARGC_REG = 0x1004 >> 2; //!< EPA of number of arguments kernel will use. 
-static const uint32_t ARGV_REG = 0x1008 >> 2; //!< EPA of arguments for kernel. 
-static const uint32_t SIGNAL_REG = 0x100c >> 2; //!< EPA of register that holds signal address. Tile will write to this address once it completes the kernel.   
+static const hb_mc_epa_t KERNEL_REG = 0x1000; //!< EPA of kernel. 
+static const hb_mc_epa_t ARGC_REG   = 0x1004; //!< EPA of number of arguments kernel will use. 
+static const hb_mc_epa_t ARGV_REG   = 0x1008; //!< EPA of arguments for kernel. 
+static const hb_mc_epa_t SIGNAL_REG = 0x100c; //!< EPA of register that holds signal address. Tile will write to this address once it completes the kernel.   
 
 
-static const uint32_t FINISH_ADDRESS = 0xC0DA; //!< EPA to which tile group sends a finish packet once it finishes executing a kernel  
-
-static awsbwhal::MemoryManager *mem_manager[1] = {(awsbwhal::MemoryManager *) 0}; /* This array has an element for every EVA <-> NPA mapping. Currently, only one mapping is supported. */
+static const hb_mc_epa_t FINISH_BASE_ADDR = 0xF000; //!< EPA to which tile group sends a finish packet once it finishes executing a kernel  
 
 static uint32_t const DRAM_SIZE = 0x80000000;
```

Comment:
I believe there is a method for getting the DRAM size.

Reply:
Yes you are right, thanks for pointing it out! I removed the parameter and used the hb_mc_config_get_dram_size() instead.